### ISSUE TITLE
[BUG] Error code like "The specified HTTP verb (GET) is not valid" would be thrown out while querying for Service Bus DLQ (Dead Letter Queue) scale status

### ISSUE THREAD

**Issue by shaykchen** [Azure/azure-sdk-for-net#36261]
### Library name and version

ServiceBus 5.8.1.0

### Describe the bug

Customer got the below exception while trying to process ServiceBus messages in the Dead letter queue:

> Error querying for Service Bus queue scale status: The specified HTTP verb (GET) is not valid. To know more visit https://aka.ms/sbResourceMgrExceptions. . TrackingId:******-****-****-****-**********_***, SystemTracker:***-******-******-******-***-001:Queue:sb-metadata-mgmt-queue-in, Timestamp:2023-04-03T07:12:24

We involved Service Bus team and they assumed the issue is due to:

> The function internal Scale Monitor is trying to get RuntimeInfo for the entity. In cx's case, this entity path is for the DLQ i.e. <queue-name>/$deadletterqueue.”

> Now RuntimeInfo is a metadata info like lockDuration, TTL, etc. However, this will not work for DLQ sub queue. As DLQ is a sub-queue for the main entity (queue/subscription), hence it doesn't contain any metadata info, hence this method will be failed if the user will try to use the DLQ path as entity.

> Nevertheless, in the latest version of the SB extension, here is the line which can cause the same issue: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusMetricsProvider.cs#L53.

> Only difference here is that it's using a latest SB package "Azure.Messaging.ServiceBus.Administration" ServiceBusAdministrationClient client, instead of old package ManagementClient :

```
 public async Task<long> GetMessageCountAsync()
        {
            long activeMessageCount = 0;
            long deadLetterCount = 0;
            string entityName = _serviceBusEntityType == ServiceBusEntityType.Queue ? "queue" : "topic";
            try
            {
                if (_serviceBusEntityType == ServiceBusEntityType.Queue)
                {
                    QueueRuntimeProperties queueRuntimeProperties = await _administrationClient.Value.GetQueueRuntimePropertiesAsync(_entityPath).ConfigureAwait(false);
                    activeMessageCount = queueRuntimeProperties.ActiveMessageCount;
                    deadLetterCount = queueRuntimeProperties.DeadLetterMessageCount;
                }
                else
                {
                    ServiceBusEntityPathHelper.ParseTopicAndSubscription(_entityPath, out string topicPath, out string subscriptionPath);

                    SubscriptionRuntimeProperties subscriptionProperties = await _administrationClient.Value.GetSubscriptionRuntimePropertiesAsync(topicPath, subscriptionPath).ConfigureAwait(false);
                    activeMessageCount = subscriptionProperties.ActiveMessageCount;
                    deadLetterCount = subscriptionProperties.DeadLetterMessageCount;
                }
            }
```
[GetQueueRuntimePropertiesAsync internally makes a EntityNameHelper.CheckValidQueueName call to just check if the entity name is valid or not, but doesn't remove the /$Deadletterqueue path]

As the _entityPath is the same path which client provided (contains DLQ sub-path), hence it'll cause the same issue.

**Discussed with Alexey in the CRI 366758391 offline and we assumed the issue could be fixed by removing “/$Deadletterqueue” from the path and using queueRuntimeProperties.DeadLetterMessageCount. So opened this issue to trace this request.**

### Expected behavior

The messages in the Service Bus queue could be processed without the above error code.

### Actual behavior

The bellow error would be thrown out in FunctionsLogs:

> Error querying for Service Bus queue scale status: The specified HTTP verb (GET) is not valid. To know more visit https://aka.ms/sbResourceMgrExceptions. . TrackingId:******-****-****-****-**********_***, SystemTracker:***-******-******-******-***-001:Queue:sb-metadata-mgmt-queue-in, Timestamp:2023-04-03T07:12:24

### Reproduction Steps

Compose an Azure Function Service Bus Queue trigger function and configured it to processing messages from the Dead Letter queue:

For example:

Queuename: "<topic name for the original topic>/Subscriptions/<subscription name>/$DeadLetterQueue

Below is the snippet of a sample code in C# to understand better.

```
public class Function
    {
        [FunctionName("Function")]
        public void Run([ServiceBusTrigger("testdeadletter/Subscriptions/deadtest/$DeadLetterQueue", Connection = "samplebus_RootManageSharedAccessKey_SERVICEBUS")]string myQueueItem, ILogger log)
        {
            log.LogInformation($"C# ServiceBus queue trigger function processed message: {myQueueItem}");
        }
    }
}
```

### Environment

Customer's repro environment is:

- Linux Elastic Premier plan 
- Function Host Framework (4.19)
- Python 3.9 
- ServiceBus 5.8.1.0
- Extension Bundle 2.23.0

**Comment by JoshLove-msft** [comment]
> Discussed with Alexey in the CRI 366758391 offline and we assumed the issue could be fixed by removing “/$Deadletterqueue” from the path and using queueRuntimeProperties.DeadLetterMessageCount. So opened this issue to trace this request.

Agreed that this sounds like the correct fix. Related feature request asking for the ability to specify that a function should listen to the DLQ - https://github.com/Azure/azure-sdk-for-net/issues/35901

**Comment by shaykchen** [comment]
Thank you so much Josh for the quick fix!
